### PR TITLE
aws-gp3 support: update vendor aws-sdk-go and terraform-provider-aws

### PR DESCRIPTION
**Pull request by mtulio**
The motivation of this PR is to update dependencies on vendors to support gp3 volumes by default, used on the [original PR to change the default IPI to gp3](https://github.com/openshift/installer/pull/5239) - and leave it ready for the installer team take the decision to the next steps.

Vendors:
- aws-sdk-go : `v1.35.37` is the minimal version to support gp3 volumes
- terraform-provider-aws (openshift's fork): the minimal validation was added, which means that not all the entire features were added on gp3 (IOPS and throughput) was supported - it is out of the scope of 5239.

Blockers / dependencies (terraform-provider-aws bump):
- [x] bump terraform-provider-aws to base release-3.1.0
https://github.com/openshift/terraform-provider-aws/pull/13
- [x] create tag `v3.1.0*` on terraform-provider-aws' fork 
[v3.1.0-openshift-1](https://github.com/openshift/terraform-provider-aws/releases/tag/v3.1.0-openshift-1)
- [x] update `go.mod`

Reference:
- `v1.35.37` was [already verified](https://github.com/openshift/terraform-provider-aws/pull/11#issuecomment-972524238) with https://github.com/openshift/installer/pull/4729 .

### CODE REVIEW

File: vendor/modules.txt [hunk 1]
```diff
@@ -3064,7 +3065,7 @@ sigs.k8s.io/yaml
 # github.com/metal3-io/cluster-api-provider-baremetal => github.com/openshift/cluster-api-provider-baremetal v0.0.0-20190821174549-a2a477909c1d
 # github.com/openshift/client-go => github.com/openshift/client-go v0.0.0-20200929181438-91d71ef2122c
 # github.com/openshift/machine-config-operator => github.com/openshift/machine-config-operator v0.0.1-0.20201009041932-4fe8559913b8
-# github.com/terraform-providers/terraform-provider-aws => github.com/openshift/terraform-provider-aws v1.60.1-0.20210622193531-7d13cfbb1a8c
+# github.com/terraform-providers/terraform-provider-aws => github.com/mtulio/terraform-provider-aws v1.60.1-0.20211111005605-66e1a671ead5
```
Comment (mtulio) [inline vendor/modules.txt:3068]: > `github.com/mtulio` 

Validation only, waiting for [PR](https://github.com/openshift/terraform-provider-aws/pull/10) review and tag creation to update here.

File: go.mod [hunk 2]
```diff
@@ -134,7 +134,7 @@ replace (
 	github.com/metal3-io/cluster-api-provider-baremetal => github.com/openshift/cluster-api-provider-baremetal v0.0.0-20190821174549-a2a477909c1d // Pin OpenShift fork
 	github.com/openshift/client-go => github.com/openshift/client-go v0.0.0-20200929181438-91d71ef2122c // Pin client-go
 	github.com/openshift/machine-config-operator => github.com/openshift/machine-config-operator v0.0.1-0.20201009041932-4fe8559913b8 // Pin MCO so it doesn't get downgraded
-	github.com/terraform-providers/terraform-provider-aws => github.com/openshift/terraform-provider-aws v1.60.1-0.20210622193531-7d13cfbb1a8c // Pin to openshift fork with tag v2.67.0-openshift-1
+	github.com/terraform-providers/terraform-provider-aws => github.com/mtulio/terraform-provider-aws v1.60.1-0.20211111005605-66e1a671ead5 // Pin to openshift fork with tag v2.67.0-openshift-2
```
Comment (mtulio) [inline go.mod:137]: > `github.com/mtulio` 

Validation only, waiting for [PR](https://github.com/openshift/terraform-provider-aws/pull/10) review and tag creation to update here.

**Comment by mtulio** [comment]
I just launch a new cluster and everything looks good:
``` bash
$ oc get co
NAME                                       VERSION                              AVAILABLE   PROGRESSING   DEGRADED   SINCE   MESSAGE
authentication                             4.10.0-0.nightly-2021-12-06-201335   True        False         False      42m     
baremetal                                  4.10.0-0.nightly-2021-12-06-201335   True        False         False      59m     
cloud-controller-manager                   4.10.0-0.nightly-2021-12-06-201335   True        False         False      61m     
cloud-credential                           4.10.0-0.nightly-2021-12-06-201335   True        False         False      61m     
cluster-api                                4.10.0-0.nightly-2021-12-06-201335   True        False         False      61m     
cluster-autoscaler                         4.10.0-0.nightly-2021-12-06-201335   True        False         False      59m     
config-operator                            4.10.0-0.nightly-2021-12-06-201335   True        False         False      61m     
console                                    4.10.0-0.nightly-2021-12-06-201335   True        False         False      44m     
csi-snapshot-controller                    4.10.0-0.nightly-2021-12-06-201335   True        False         False      60m     
dns                                        4.10.0-0.nightly-2021-12-06-201335   True        False         False      59m     
etcd                                       4.10.0-0.nightly-2021-12-06-201335   True        False         False      59m     
image-registry                             4.10.0-0.nightly-2021-12-06-201335   True        False         False      51m     
ingress                                    4.10.0-0.nightly-2021-12-06-201335   True        False         False      51m     
insights                                   4.10.0-0.nightly-2021-12-06-201335   True        False         False      54m     
kube-apiserver                             4.10.0-0.nightly-2021-12-06-201335   True        False         False      57m     
kube-controller-manager                    4.10.0-0.nightly-2021-12-06-201335   True        False         False      58m     
kube-scheduler                             4.10.0-0.nightly-2021-12-06-201335   True        False         False      58m     
kube-storage-version-migrator              4.10.0-0.nightly-2021-12-06-201335   True        False         False      61m     
machine-api                                4.10.0-0.nightly-2021-12-06-201335   True        False         False      57m     
machine-approver                           4.10.0-0.nightly-2021-12-06-201335   True        False         False      60m     
machine-config                             4.10.0-0.nightly-2021-12-06-201335   True        False         False      59m     
marketplace                                4.10.0-0.nightly-2021-12-06-201335   True        False         False      60m     
monitoring                                 4.10.0-0.nightly-2021-12-06-201335   True        False         False      49m     
network                                    4.10.0-0.nightly-2021-12-06-201335   True        False         False      61m     
node-tuning                                4.10.0-0.nightly-2021-12-06-201335   True        False         False      60m     
openshift-apiserver                        4.10.0-0.nightly-2021-12-06-201335   True        False         False      51m     
openshift-controller-manager               4.10.0-0.nightly-2021-12-06-201335   True        False         False      59m     
openshift-samples                          4.10.0-0.nightly-2021-12-06-201335   True        False         False      50m     
operator-lifecycle-manager                 4.10.0-0.nightly-2021-12-06-201335   True        False         False      60m     
operator-lifecycle-manager-catalog         4.10.0-0.nightly-2021-12-06-201335   True        False         False      60m     
operator-lifecycle-manager-packageserver   4.10.0-0.nightly-2021-12-06-201335   True        False         False      51m     
service-ca                                 4.10.0-0.nightly-2021-12-06-201335   True        False         False      61m     
storage                                    4.10.0-0.nightly-2021-12-06-201335   True        False         False      60m   
```

**Comment by mtulio** [comment]
@staebler @jstuever @jhixson74  continuing the change from [terraform-provider-aws](https://github.com/openshift/terraform-provider-aws/pull/13) , it's ready for review. ptal?

**Comment by staebler** [comment]
Need to run `go mod tidy`.
/test e2e-gcp

**Review by staebler — APPROVED** [review]
/lgtm
/approve

**Comment by mtulio** [comment]
I will review the tests failing and report here if it's flakes.
/hold

**Comment by gregsheremeta** [comment]
I am so excited about this! Do we have any idea about which version of OCP will get this? Any chance of a backport to 4.9?

**Comment by staebler** [comment]
> I am so excited about this! Do we have any idea about which version of OCP will get this? Any chance of a backport to 4.9?

4.10. A backport is unlikely.

**Comment by mtulio** [comment]
I just launch a new cluster with the latest version[1]. I am ok to remove WIP labels with the CI is fine.

Looking the failed jobs, I am concerned only with `okd-e2e-aws` - specific for [this error](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_installer/5373/pull-ci-openshift-installer-master-okd-e2e-aws/1469294639012909056#1:build-log.txt%3A197). I am not sure if it's directly related with this PR. wdyt @staebler @rvanderp3   ?

[1]
``` shell
$ oc get clusterversion
NAME      VERSION                              AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.10.0-0.nightly-2021-12-10-123024   True        False         11m     Cluster version is 4.10.0-0.nightly-2021-12-10-123024

$ oc get co
NAME                                       VERSION                              AVAILABLE   PROGRESSING   DEGRADED   SINCE   MESSAGE
authentication                             4.10.0-0.nightly-2021-12-10-123024   True        False         False      11m     
baremetal                                  4.10.0-0.nightly-2021-12-10-123024   True        False         False      29m     
cloud-controller-manager                   4.10.0-0.nightly-2021-12-10-123024   True        False         False      31m     
cloud-credential                           4.10.0-0.nightly-2021-12-10-123024   True        False         False      30m     
cluster-api                                4.10.0-0.nightly-2021-12-10-123024   True        False         False      30m     
cluster-autoscaler                         4.10.0-0.nightly-2021-12-10-123024   True        False         False      29m     
config-operator                            4.10.0-0.nightly-2021-12-10-123024   True        False         False      30m     
console                                    4.10.0-0.nightly-2021-12-10-123024   True        False         False      17m     
csi-snapshot-controller                    4.10.0-0.nightly-2021-12-10-123024   True        False         False      30m     
dns                                        4.10.0-0.nightly-2021-12-10-123024   True        False         False      29m     
etcd                                       4.10.0-0.nightly-2021-12-10-123024   True        False         False      29m     
image-registry                             4.10.0-0.nightly-2021-12-10-123024   True        False         False      21m     
ingress                                    4.10.0-0.nightly-2021-12-10-123024   True        False         False      23m     
insights                                   4.10.0-0.nightly-2021-12-10-123024   True        False         False      24m     
kube-apiserver                             4.10.0-0.nightly-2021-12-10-123024   True        False         False      23m     
kube-controller-manager                    4.10.0-0.nightly-2021-12-10-123024   True        False         False      29m     
kube-scheduler                             4.10.0-0.nightly-2021-12-10-123024   True        False         False      29m     
kube-storage-version-migrator              4.10.0-0.nightly-2021-12-10-123024   True        False         False      30m     
machine-api                                4.10.0-0.nightly-2021-12-10-123024   True        False         False      26m     
machine-approver                           4.10.0-0.nightly-2021-12-10-123024   True        False         False      29m     
machine-config                             4.10.0-0.nightly-2021-12-10-123024   True        False         False      21m     
marketplace                                4.10.0-0.nightly-2021-12-10-123024   True        False         False      29m     
monitoring                                 4.10.0-0.nightly-2021-12-10-123024   True        False         False      18m     
network                                    4.10.0-0.nightly-2021-12-10-123024   True        False         False      31m     
node-tuning                                4.10.0-0.nightly-2021-12-10-123024   True        False         False      30m     
openshift-apiserver                        4.10.0-0.nightly-2021-12-10-123024   True        False         False      22m     
openshift-controller-manager               4.10.0-0.nightly-2021-12-10-123024   True        False         False      30m     
openshift-samples                          4.10.0-0.nightly-2021-12-10-123024   True        False         False      21m     
operator-lifecycle-manager                 4.10.0-0.nightly-2021-12-10-123024   True        False         False      30m     
operator-lifecycle-manager-catalog         4.10.0-0.nightly-2021-12-10-123024   True        False         False      30m     
operator-lifecycle-manager-packageserver   4.10.0-0.nightly-2021-12-10-123024   True        False         False      21m     
service-ca                                 4.10.0-0.nightly-2021-12-10-123024   True        False         False      30m     
storage                                    4.10.0-0.nightly-2021-12-10-123024   True        False         False      30m   
```

**Comment by rvanderp3** [comment]
> I just launch a new cluster with the latest version[1]. I am ok to remove WIP labels if CI is fine.
> 
> Looking the failed jobs, I am concerned only with `okd-e2e-aws` - specific for [this error](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_installer/5373/pull-ci-openshift-installer-master-okd-e2e-aws/1469294639012909056#1:build-log.txt%3A197). I am not sure if it's directly related with this PR. wdyt @staebler @rvanderp3 ?
> 
> [1]
> 
> ```shell
> $ oc get clusterversion
> NAME      VERSION                              AVAILABLE   PROGRESSING   SINCE   STATUS
> version   4.10.0-0.nightly-2021-12-10-123024   True        False         11m     Cluster version is 4.10.0-0.nightly-2021-12-10-123024
> 
> $ oc get co
> NAME                                       VERSION                              AVAILABLE   PROGRESSING   DEGRADED   SINCE   MESSAGE
> authentication                             4.10.0-0.nightly-2021-12-10-123024   True        False         False      11m     
> baremetal                                  4.10.0-0.nightly-2021-12-10-123024   True        False         False      29m     
> cloud-controller-manager                   4.10.0-0.nightly-2021-12-10-123024   True        False         False      31m     
> cloud-credential                           4.10.0-0.nightly-2021-12-10-123024   True        False         False      30m     
> cluster-api                                4.10.0-0.nightly-2021-12-10-123024   True        False         False      30m     
> cluster-autoscaler                         4.10.0-0.nightly-2021-12-10-123024   True        False         False      29m     
> config-operator                            4.10.0-0.nightly-2021-12-10-123024   True        False         False      30m     
> console                                    4.10.0-0.nightly-2021-12-10-123024   True        False         False      17m     
> csi-snapshot-controller                    4.10.0-0.nightly-2021-12-10-123024   True        False         False      30m     
> dns                                        4.10.0-0.nightly-2021-12-10-123024   True        False         False      29m     
> etcd                                       4.10.0-0.nightly-2021-12-10-123024   True        False         False      29m     
> image-registry                             4.10.0-0.nightly-2021-12-10-123024   True        False         False      21m     
> ingress                                    4.10.0-0.nightly-2021-12-10-123024   True        False         False      23m     
> insights                                   4.10.0-0.nightly-2021-12-10-123024   True        False         False      24m     
> kube-apiserver                             4.10.0-0.nightly-2021-12-10-123024   True        False         False      23m     
> kube-controller-manager                    4.10.0-0.nightly-2021-12-10-123024   True        False         False      29m     
> kube-scheduler                             4.10.0-0.nightly-2021-12-10-123024   True        False         False      29m     
> kube-storage-version-migrator              4.10.0-0.nightly-2021-12-10-123024   True        False         False      30m     
> machine-api                                4.10.0-0.nightly-2021-12-10-123024   True        False         False      26m     
> machine-approver                           4.10.0-0.nightly-2021-12-10-123024   True        False         False      29m     
> machine-config                             4.10.0-0.nightly-2021-12-10-123024   True        False         False      21m     
> marketplace                                4.10.0-0.nightly-2021-12-10-123024   True        False         False      29m     
> monitoring                                 4.10.0-0.nightly-2021-12-10-123024   True        False         False      18m     
> network                                    4.10.0-0.nightly-2021-12-10-123024   True        False         False      31m     
> node-tuning                                4.10.0-0.nightly-2021-12-10-123024   True        False         False      30m     
> openshift-apiserver                        4.10.0-0.nightly-2021-12-10-123024   True        False         False      22m     
> openshift-controller-manager               4.10.0-0.nightly-2021-12-10-123024   True        False         False      30m     
> openshift-samples                          4.10.0-0.nightly-2021-12-10-123024   True        False         False      21m     
> operator-lifecycle-manager                 4.10.0-0.nightly-2021-12-10-123024   True        False         False      30m     
> operator-lifecycle-manager-catalog         4.10.0-0.nightly-2021-12-10-123024   True        False         False      30m     
> operator-lifecycle-manager-packageserver   4.10.0-0.nightly-2021-12-10-123024   True        False         False      21m     
> service-ca                                 4.10.0-0.nightly-2021-12-10-123024   True        False         False      30m     
> storage                                    4.10.0-0.nightly-2021-12-10-123024   True        False         False      30m   
> ```

@mtulio this looks like a possible flake similar to what was reported in https://github.com/hashicorp/terraform-provider-aws/issues/16142 .  I'm not sure it's related but let's see how the currently running okd install goes.

**Comment by staebler** [comment]
> @mtulio this looks like a possible flake similar to what was reported in [hashicorp/terraform-provider-aws#16142](https://github.com/hashicorp/terraform-provider-aws/issues/16142) . I'm not sure it's related but let's see how the currently running okd install goes.

Yes, I concur.

**Pull request merged**